MobileWeather/app/src/main/java/com/sdl/mobileweather/activity/MainActivity.java

```diff
@@ -139,14 +144,28 @@ else if ((getResources().getString(R.string.drawer_item_about)).equals(item)){
 	private void checkForCrashes() {}
 
 	private void checkForUpdates() {}
-	
+
+	private boolean checkPermission() {
+		return PackageManager.PERMISSION_GRANTED == ContextCompat.checkSelfPermission(getApplicationContext(), BLUETOOTH_CONNECT);
+	}
+
+	private void requestPermission() {
+		ActivityCompat.requestPermissions(this, new String[]{BLUETOOTH_CONNECT}, BLUETOOTH_REQUEST_CODE);
+	}
+
 	@Override
     protected void onCreate(Bundle savedInstanceState) {
     	Log.v(SdlApplication.TAG, "onCreate main");
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
-        
-        
+
+		if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S && !checkPermission()) {
+			requestPermission();
+			return;
+		}
+
+		SdlReceiver.queryForConnectedService(this);
```

Comment:
We do not want to call `SdlReceiver.queryForConnectedService(this);` here. If you look at the flow of the app, it gets called after the location permission is accepted. I would recommend that we tie in existing logic and request the nearby device permission after the location permission, possibly in `SdlApplication.startService` before it calls `startSdlProxyService();`. There are a couple of different ways to do it.